src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1963,7 +1963,48 @@ public TryWriteInterpolatedStringHandler(int literalLength, int formattedCount,
             /// <summary>Writes the specified string to the handler.</summary>
             /// <param name="value">The string to write.</param>
             /// <returns>true if the value could be formatted to the span; otherwise, false.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public bool AppendLiteral(string value)
+            {
+                // See comment on inlining and special-casing in DefaultInterpolatedStringHandler.AppendLiteral.
+
+                if (value.Length == 1)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < (uint)destination.Length)
+                    {
+                        destination[pos] = value[0];
+                        _pos = pos + 1;
+                        return true;
+                    }
+
+                    return Fail();
+                }
+
+                if (value.Length == 2)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < destination.Length - 1)
+                    {
+                        Unsafe.WriteUnaligned(
+                            ref MemoryMarshal.GetReference(MemoryMarshal.AsBytes(destination)),
```

Comment:
`MemoryMarshal.AsBytes(destination)` can throw. And it looks like this code is always writing to the start of _destination_ rather than offset _pos_. Prefer instead: `Unsafe.WriteUnaligned(ref Unsafe.As<char, byte>(ref Unsafe.Add(ref MemoryMarshal.GetReference(destination), pos)), ...)`.

Or, if we're willing to incur two bounds checks instead of one:

```cs
Span<char> sliced = _destination.Slice(pos); // first bounds check
if (sliced.Length >= 2) // second bounds check
{
    ref byte dest = ref Unsafe.As<char, byte>(ref pos[0]); // bounds check against pos[0] should be elided
    Unsafe.WriteUnaligned<int>(ref dest, /* ... */);
}
```

(Same comment in other code block.)

Reply:
> And it looks like this code is always writing to the start of destination rather than offset pos

Yeah, that was me pushing the revised commit before running the tests.